addons/test_event_full/tests/test_website_event.py

```diff
@@ -0,0 +1,58 @@
+from datetime import datetime, timedelta
+from itertools import combinations
+
+from odoo import fields
+from odoo.addons.test_event_full.tests.common import TestEventFullCommon
+from odoo.tests.common import users
+
+
+class TestWebsiteEvent(TestEventFullCommon):
+
+    @users('user_event_web_manager')
+    def test_seo_data(self):
+        """Test SEO data for submenus on event website page"""
+
+        # Create an event
+        event = self.env['event.event'].create({
+            'name': 'TestEvent',
+            'date_begin': fields.Datetime.to_string(datetime.today() + timedelta(days=1)),
+            'date_end': fields.Datetime.to_string(datetime.today() + timedelta(days=15)),
+            'website_menu': True,
+            'booth_menu': True,
+            'exhibitor_menu': True,
+            'community_menu': True,
+            'website_track': True,
+            'website_track_proposal': True,
+        })
+
+        self.assertFalse(event.website_meta_title, 'Event should initially have no meta title')
+        event.write({
+            'website_meta_title': 'info',
+        })
+        self.assertTrue(event.website_meta_title, 'Event should have a meta title after writing')
+
+        # Run SEO tests and store view models for different pages
+        view_model_booth = self.assert_seo_data(event, 'Get A Booth', 'booth')
+        view_model_exhibitor = self.assert_seo_data(event, 'Exhibitors', 'exhibitors')
+        view_model_community = self.assert_seo_data(event, 'Community', 'community')
+        view_model_track = self.assert_seo_data(event, 'Talks', 'talks')
+        view_model_agenda = self.assert_seo_data(event, 'Agenda', 'agenda')
+        view_model_track_proposal = self.assert_seo_data(event, 'Talk Proposals', 'talk proposals')
+
+        # SEO data test of submenus
+        view_models = [
+            (view_model_booth, 'Booth page'),
+            (view_model_exhibitor, 'Exhibitor page'),
+            (view_model_community, 'Community page'),
+            (view_model_track, 'Talks page'),
+            (view_model_agenda, 'Agenda page'),
+            (view_model_track_proposal, 'Talk Proposals page')
+        ]
+
+        # Compare each view model with every other view model to ensure their meta titles are different
+        for (model1, name1), (model2, name2) in combinations(view_models, 2):
+            self.assertNotEqual(
+                model1.website_meta_title,
+                model2.website_meta_title,
+                f'{name1} and {name2} should have different meta titles'
+            )
```

Comment:
Would say globally we can trust that the views will have their own meta title, that's more of a website test.

What we can check is that every menu has its own view, and then maybe sanity check that the meta_title is what we expect for that menu (mostly to help show why the individual views are needed.

so:
- after the event is created check that each menu has its own view (you can use `track_menu_ids.view`, no need for search)
- assign a website meta title
- check that the meta title is as expected for each view

Optionally, we could make this an http case, and actually visit the urls to make sure we have the right meta title at rendering (lxml parse the result + xpath search). Or at least that we pass in the right "seo_object" (patch the route to get the lazy render object it returns, and check the render values). But it may be overkill for testing simple fetching logic ^^

Reply:
Hello @reth-odoo 
Is this what you meant for the test cases? I pushed the updated code.
Thanks :blush: